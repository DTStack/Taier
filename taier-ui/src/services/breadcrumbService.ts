import { CATALOGUE_TYPE } from '@/constant';
import { TreeViewUtil } from '@dtinsight/molecule/esm/common/treeUtil';
import type molecule from '@dtinsight/molecule';
import type { UniqueId } from '@dtinsight/molecule/esm/common/types';
import type { IBreadcrumbItemProps } from '@dtinsight/molecule/esm/components';
import { catalogueService } from '.';

interface IBreadcrumbService {
    /**
     * 根据目录树 id 获取父节点并生成面包屑
     */
    getBreadcrumb: (id: UniqueId) => IBreadcrumbItemProps[];
}

export default class BreadcrumbService implements IBreadcrumbService {
    private hashTree: TreeViewUtil<molecule.model.IFolderTreeNodeProps> | null = null;
    constructor() {
        catalogueService.onUpdate(() => {
            this.updateTree();
        });
    }

    private updateTree = () => {
        const root = catalogueService.getRootFolder(CATALOGUE_TYPE.TASK);
        if (root) {
            this.hashTree = new TreeViewUtil<molecule.model.IFolderTreeNodeProps>(root);
        }
    };

    public getBreadcrumb = (id: UniqueId) => {
        if (this.hashTree) {
            const stack = [this.hashTree.getHashMap(id)];
            const res: IBreadcrumbItemProps[] = [];
            while (stack.length) {
                const hash = stack.pop();
                if (hash) {
                    res.push({
                        id: hash.node.id,
                        name: hash.node.name,
                    });

                    if (hash.parent) {
                        stack.push(this.hashTree.getHashMap(hash.parent));
                    }
                }
            }

            return res.reverse();
        }
        return [];
    };
}
