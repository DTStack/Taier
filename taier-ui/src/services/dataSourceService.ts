/*
 * Licensed to the Apache Software Foundation (ASF) under one
 * or more contributor license agreements.  See the NOTICE file
 * distributed with this work for additional information
 * regarding copyright ownership.  The ASF licenses this file
 * to you under the Apache License, Version 2.0 (the
 * "License"); you may not use this file except in compliance
 * with the License.  You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import { IDataSourceProps } from '@/interface';
import { Component } from '@dtinsight/molecule/esm/react';
import Base64 from 'base-64';
import api from '@/api';

export interface IDataSourceState {
    dataSource: IDataSourceProps[];
}

interface IDataSourceService {
    getDataSource: () => IDataSourceProps[];
    reloadDataSource: () => void;
}

export default class DataSourceService extends Component<IDataSourceState> implements IDataSourceService {
    protected state: IDataSourceState = {
        dataSource: [],
    };

    constructor() {
        super();
        this.queryDataSource();
    }

    private queryDataSource = () => {
        api.getAllDataSource({}).then((res) => {
            if (res.code === 1) {
                const nextData: IDataSourceProps[] = ((res.data as IDataSourceProps[]) || []).map((ele) => {
                    const canConvertLinkJson =
                        ele.linkJson && !ele.linkJson.includes('{') && !ele.linkJson.includes('}');

                    return {
                        ...ele,
                        linkJson: canConvertLinkJson ? Base64.decode(ele.linkJson!) : ele.linkJson,
                    };
                });

                this.setState({
                    dataSource: nextData,
                });
            }
        });
    };

    getDataSource = () => {
        return this.state.dataSource || [];
    };

    reloadDataSource = () => {
        this.queryDataSource();
    };
}
