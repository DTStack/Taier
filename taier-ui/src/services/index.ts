import { container } from 'tsyringe';
import EditorActionBarService from './editorActionBarService';
import ExecuteService from './executeService';
import CatalogueService from './catalogueService';
import BreadcrumbService from './breadcrumbService';
import RightBarService from './rightBarService';
import TaskRenderService from './taskRenderService';
import DataSourceService from './dataSourceService';
import TaskParamsService from './taskParamsService';

const editorActionBarService = container.resolve(EditorActionBarService);
const executeService = container.resolve(ExecuteService);
const catalogueService = container.resolve(CatalogueService);
const breadcrumbService = container.resolve(BreadcrumbService);
const rightBarService = container.resolve(RightBarService);
const taskRenderService = container.resolve(TaskRenderService);
const dataSourceService = container.resolve(DataSourceService);
const taskParamsService = container.resolve(TaskParamsService);

export {
    editorActionBarService,
    catalogueService,
    executeService,
    breadcrumbService,
    rightBarService,
    taskRenderService,
    dataSourceService,
    taskParamsService,
};
