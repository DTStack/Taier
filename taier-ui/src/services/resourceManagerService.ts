/*
 * Licensed to the Apache Software Foundation (ASF) under one
 * or more contributor license agreements.  See the NOTICE file
 * distributed with this work for additional information
 * regarding copyright ownership.  The ASF licenses this file
 * to you under the Apache License, Version 2.0 (the
 * "License"); you may not use this file except in compliance
 * with the License.  You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import { FolderTreeService } from '@dtinsight/molecule/esm/services';

let resourceManagerTree;
class ResourceManagerTree extends FolderTreeService {
    /**
     * 判断该节点是否为 file
     * @description 资源管理的所有节点，文件夹节点的 id 为 `xxx-folder`，只有文件节点的 id 是可以直接通过服务端接口数据的 id 获取数据
     * @link {catalogueService#L90-106}
     */
    checkNotDir = (id: number | string) => {
        if (typeof id === 'number') {
            const node = this.get(id);
            return node ? Promise.resolve() : Promise.reject(new Error('请选择具体文件, 而非文件夹'));
        }

        return Promise.reject(new Error('请选择具体文件, 而非文件夹'));
    };
}

if (!resourceManagerTree) {
    resourceManagerTree = new ResourceManagerTree();
}

export default resourceManagerTree as ResourceManagerTree;
