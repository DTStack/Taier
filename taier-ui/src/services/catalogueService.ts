import api from '@/api';
import { CATALOGUE_TYPE, MENU_TYPE_ENUM } from '@/constant';
import type { CatalogueDataProps } from '@/interface';
import { getTenantId, getUserId } from '@/utils';
import { fileIcon } from '@/utils/extensions';
import molecule from '@dtinsight/molecule';
import { GlobalEvent } from '@dtinsight/molecule/esm/common/event';
import { FileTypes, TreeNodeModel } from '@dtinsight/molecule/esm/model';
import { singleton } from 'tsyringe';
import functionManagerService from './functionManagerService';
import resourceManagerService from './resourceManagerService';

interface ICatalogueService {
    /**
     * 加载目录树的根目录
     */
    loadRootFolder: () => void;
    /**
     * 获取目录树的根目录
     */
    getRootFolder: (source: CATALOGUE_TYPE) => molecule.model.IFolderTreeNodeProps | undefined;
    /**
     * 获取目录树的子节点
     * @param node 更新当前 Node 节点的子节点，不改变 Node 节点
     */
    loadTreeNode: (node: Partial<Pick<CatalogueDataProps, 'id' | 'catalogueType'>>, source: CATALOGUE_TYPE) => void;
    /**
     * 目录树更新监听事件
     */
    onUpdate: (callback: () => void) => void;
}

export enum CatalogueEventKind {
    onUpdate = 'catalogue.update',
}

@singleton()
export default class CatalogueService extends GlobalEvent implements ICatalogueService {
    /**
     * 异步加载目录树节点
     */
    private getCatalogueViaNode = async (
        node: Partial<Pick<CatalogueDataProps, 'id' | 'catalogueType'>>
    ): Promise<CatalogueDataProps | undefined> => {
        if (!node) throw new Error('[getCatalogueViaNode]: failed to get catelogue');
        const res = await api.getOfflineCatalogue({
            isGetFile: true,
            nodePid: node.id,
            catalogueType: node.catalogueType,
            userId: getUserId(),
            tenantId: getTenantId(),
        });
        if (res.code === 1) {
            return res.data;
        }
        return undefined;
    };

    private getRootFolderViaSource = (data: CatalogueDataProps[], source: CATALOGUE_TYPE) => {
        switch (source) {
            case CATALOGUE_TYPE.TASK: {
                return data.find((item) => item.catalogueType === MENU_TYPE_ENUM.TASK);
            }

            case CATALOGUE_TYPE.RESOURCE: {
                return data.find((item) => item.catalogueType === MENU_TYPE_ENUM.RESOURCE);
            }

            case CATALOGUE_TYPE.FUNCTION: {
                return data.find((item) => item.catalogueType === MENU_TYPE_ENUM.FUNCTION);
            }
            default:
                return undefined;
        }
    };

    /**
     * Only transform current catalogue to treeNodeModel, ignore children
     */
    private transformCatalogueToTree = (catalogue: CatalogueDataProps | undefined, source: CATALOGUE_TYPE) => {
        if (!catalogue) return;
        const folderType = ['folder', 'catalogue'];
        switch (source) {
            case CATALOGUE_TYPE.RESOURCE: {
                const fileType = folderType.includes(catalogue.type) ? FileTypes.Folder : FileTypes.File;

                const id = fileType === FileTypes.File ? catalogue.id : `${catalogue.id}-folder`;
                return new TreeNodeModel({
                    // prevent same id between folder and file
                    id,
                    name: catalogue.name,
                    fileType,
                    icon: fileIcon(catalogue.resourceType, source),
                    isLeaf: fileType === FileTypes.File,
                    data: catalogue,
                    children: resourceManagerService.get(id)?.children || [],
                });
            }

            case CATALOGUE_TYPE.FUNCTION: {
                const { type, name } = catalogue;
                const fileType = folderType.includes(type) ? FileTypes.Folder : FileTypes.File;

                // Because of the same id in different levels, so we should set another uniq id for each tree node
                const id = `${catalogue.id}-${folderType.includes(type) ? 'folder' : 'file'}`;
                return new TreeNodeModel({
                    id,
                    name,
                    location: name,
                    fileType,
                    isLeaf: fileType === FileTypes.File,
                    data: catalogue,
                    icon: fileIcon(catalogue.taskType, source),
                    children: functionManagerService.get(id)?.children || [],
                });
            }

            case CATALOGUE_TYPE.TASK: {
                const fileType = folderType.includes(catalogue.type) ? FileTypes.Folder : FileTypes.File;

                // prevent same id between folder and file
                const id = fileType === FileTypes.File ? catalogue.id : `${catalogue.id}-folder`;
                return new TreeNodeModel({
                    id,
                    name: catalogue.name,
                    location: catalogue.name,
                    fileType,
                    icon: fileIcon(catalogue.taskType, source),
                    isLeaf: fileType === FileTypes.File,
                    data: catalogue,
                    children: molecule.folderTree.get(id)?.children || [],
                });
            }

            default:
                return undefined;
        }
    };

    private getServiceBySource = (source: CATALOGUE_TYPE) => {
        switch (source) {
            case CATALOGUE_TYPE.TASK:
                return molecule.folderTree;
            case CATALOGUE_TYPE.FUNCTION:
                return functionManagerService;
            case CATALOGUE_TYPE.RESOURCE:
                return resourceManagerService;
            default:
                return null;
        }
    };

    /**
     * @param node 更新当前 Node 节点的子节点，不改变 Node 节点
     */
    public loadTreeNode = async (
        node: Partial<Pick<CatalogueDataProps, 'id' | 'catalogueType'>>,
        source: CATALOGUE_TYPE
    ) => {
        const data = await this.getCatalogueViaNode(node);
        if (data) {
            const childrenNodes = <molecule.model.TreeNodeModel[]>(
                (data?.children?.map((child) => this.transformCatalogueToTree(child, source)) || []).filter(Boolean)
            );
            const service = this.getServiceBySource(source);
            const treeNode = service?.get(`${node.id!}-folder`);
            if (treeNode) {
                service?.update({
                    ...treeNode,
                    children: childrenNodes,
                });

                this.emit(CatalogueEventKind.onUpdate);
                return service?.get(treeNode.id);
            }
        }
    };

    public getRootFolder = (source: CATALOGUE_TYPE) => {
        const service = this.getServiceBySource(source);
        switch (source) {
            case CATALOGUE_TYPE.TASK:
            case CATALOGUE_TYPE.FUNCTION:
                return service?.getState().folderTree?.data?.[0];
            case CATALOGUE_TYPE.RESOURCE:
                return service?.getState().folderTree?.data?.[0]?.children?.[0];
            default:
                return undefined;
        }
    };

    public loadRootFolder = () => {
        this.getCatalogueViaNode({ id: 0 }).then((res) => {
            if (!res || !res.children) {
                return;
            }
            const { children } = res;

            const taskData = this.getRootFolderViaSource(children, CATALOGUE_TYPE.TASK);
            const resourceData = this.getRootFolderViaSource(children, CATALOGUE_TYPE.RESOURCE);
            const funcData = this.getRootFolderViaSource(children, CATALOGUE_TYPE.FUNCTION);

            // 更新资源目录树
            if (resourceData) {
                const resourceRoot = resourceData;
                const resourceNode = this.transformCatalogueToTree(resourceRoot, CATALOGUE_TYPE.RESOURCE)!;
                const childrenNodes =
                    resourceRoot.children?.map(
                        (child) => this.transformCatalogueToTree(child, CATALOGUE_TYPE.RESOURCE)!
                    ) || [];

                // set a root folder
                resourceNode.fileType = FileTypes.RootFolder;
                // put children to root folder
                resourceNode.children = childrenNodes;
                resourceManagerService.add(resourceNode);
            }

            // 更新函数目录树
            if (funcData) {
                const funcRoot = funcData;
                const functionNode = this.transformCatalogueToTree(funcRoot, CATALOGUE_TYPE.FUNCTION)!;
                const childrenNodes =
                    funcRoot.children
                        // there is a system function in the children node of root folder, we'd better to filter it
                        ?.filter((child) => child.name !== '系统函数')
                        .map((child) => this.transformCatalogueToTree(child, CATALOGUE_TYPE.FUNCTION)!) || [];

                // set a root folder
                functionNode.fileType = FileTypes.RootFolder;
                // put children to root folder
                functionNode.children = childrenNodes;
                functionManagerService.add(functionNode);

                // // sql 节点必存在 catalogueType，对所有的节点都求一遍子树
                // const SqlNodes =
                // 	funcRoot?.children?.filter((child: any) => child.catalogueType) || [];
                // SqlNodes.forEach((sqlNode) => {
                // 	loadTreeNode(sqlNode, CATELOGUE_TYPE.FUNCTION);
                // });
            }

            // 任务开发根目录
            if (taskData) {
                const taskRootFolder = taskData?.children?.[0];
                if (taskRootFolder) {
                    const taskNode = this.transformCatalogueToTree(taskRootFolder, CATALOGUE_TYPE.TASK)!;

                    taskNode.fileType = FileTypes.RootFolder;
                    molecule.folderTree.add(taskNode);

                    // 获取当前根目录的下级目录，确保打开 Explorer 有数据展示
                    this.loadTreeNode(taskRootFolder, CATALOGUE_TYPE.TASK);
                }
            }
        });
    };

    public onUpdate = (callback: () => void) => {
        this.subscribe(CatalogueEventKind.onUpdate, callback);
    };
}
