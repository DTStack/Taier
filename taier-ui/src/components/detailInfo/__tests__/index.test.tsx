import { CATALOGUE_TYPE } from '@/constant';
import { dataSourceService } from '@/services';
import { render } from '@testing-library/react';
import DetailInfo, { DetailInfoModal } from '..';

jest.mock('@/context', () => {
    const react = jest.requireActual('react');
    return react.createContext({
        supportJobTypes: [{ key: 1, value: 'spark' }],
    });
});

jest.mock('@/services', () => {
    return {
        dataSourceService: {
            getDataSource: jest.fn(),
        },
    };
});

describe('Test DetailInfo Component', () => {
    it('Should match snapshots', () => {
        expect(
            render(<DetailInfoModal type={CATALOGUE_TYPE.TASK} title="任务详情" visible loading={false} />).asFragment()
        ).toMatchSnapshot();
    });

    describe('Test DetailInfo Component', () => {
        it('Should match snapshots', () => {
            (dataSourceService.getDataSource as jest.Mock).mockReset().mockImplementation(() => [
                {
                    dataInfoId: 1,
                    dataName: 'dataName',
                    dataType: 1,
                },
            ]);

            expect(
                render(
                    <DetailInfo
                        type={CATALOGUE_TYPE.TASK}
                        data={{
                            name: 'test',
                            taskType: 1,
                            datasourceId: 1,
                            componentVersion: '1.12',
                            gmtCreate: new Date('2022-01-01').valueOf(),
                            gmtModified: new Date('2022-01-01').valueOf(),
                            taskDesc: 'test',
                        }}
                    />
                ).asFragment()
            ).toMatchSnapshot();

            expect(
                render(
                    <DetailInfo
                        type={CATALOGUE_TYPE.FUNCTION}
                        data={{
                            name: 'test',
                            className: 'className',
                            sqlText: 'sqlText',
                            purpose: 'purpose',
                            commandFormate: 'commandFormate',
                            paramDesc: 'paramDesc',
                            gmtCreate: new Date('2022-01-01').valueOf(),
                            gmtModified: new Date('2022-01-01').valueOf(),
                        }}
                    />
                ).asFragment()
            ).toMatchSnapshot();

            expect(
                render(
                    <DetailInfo
                        type={CATALOGUE_TYPE.RESOURCE}
                        data={{
                            resourceName: 'test',
                            resourceDesc: 'resourceDesc',
                            url: 'url',
                            createUser: {
                                userName: 'userName',
                            },
                            gmtCreate: new Date('2022-01-01').valueOf(),
                            gmtModified: new Date('2022-01-01').valueOf(),
                        }}
                    />
                ).asFragment()
            ).toMatchSnapshot();

            expect(
                render(
                    <DetailInfo
                        type="dataSource"
                        data={{
                            dataInfoId: 17,
                            dataName: 'mysql_test',
                            dataType: 'MySQL',
                            dataVersion: null,
                            dataDesc: '',
                            linkJson: JSON.stringify({
                                jdbcUrl: 'jdbc:mysql://',
                                username: 'dtstack',
                            }),
                            status: 1,
                            isMeta: 0,
                            gmtModified: 1667358042000,
                            isImport: null,
                            schemaName: '',
                            dataTypeCode: 1,
                        }}
                    />
                ).asFragment()
            ).toMatchSnapshot();

            expect(
                render(
                    <DetailInfo
                        type="taskJob"
                        data={{
                            taskName: 'test',
                            jobId: 'jobId',
                            taskType: 1,
                            status: 1,
                            taskPeriodId: 0,
                            cycTime: 1667358042000,
                        }}
                    />
                ).asFragment()
            ).toMatchSnapshot();
        });
    });
});
