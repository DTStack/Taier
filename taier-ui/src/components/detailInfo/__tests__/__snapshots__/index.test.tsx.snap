// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test DetailInfo Component Should match snapshots 1`] = `
<DocumentFragment>
  <p
    data-testid="antd-mock-Modal-title"
  >
    任务详情
  </p>
  <div
    class="ant-spin-nested-loading"
  >
    <div
      class="ant-spin-container"
    >
      系统异常
    </div>
  </div>
  <button
    class="ant-btn ant-btn-default"
    type="button"
  >
    <span>
      关 闭
    </span>
  </button>
</DocumentFragment>
`;

exports[`Test DetailInfo Component Test DetailInfo Component Should match snapshots 1`] = `
<DocumentFragment>
  <div
    class="ant-descriptions ant-descriptions-small ant-descriptions-bordered dt-taskinfo"
  >
    <div
      class="ant-descriptions-view"
    >
      <table>
        <tbody>
          <tr
            class="ant-descriptions-row"
          >
            <th
              class="ant-descriptions-item-label"
              colspan="1"
            >
              <span>
                任务名称
              </span>
            </th>
            <td
              class="ant-descriptions-item-content"
              colspan="1"
            >
              <span>
                test
              </span>
            </td>
          </tr>
          <tr
            class="ant-descriptions-row"
          >
            <th
              class="ant-descriptions-item-label"
              colspan="1"
            >
              <span>
                任务类型
              </span>
            </th>
            <td
              class="ant-descriptions-item-content"
              colspan="1"
            >
              <span>
                spark
              </span>
            </td>
          </tr>
          <tr
            class="ant-descriptions-row"
          >
            <th
              class="ant-descriptions-item-label"
              colspan="1"
            >
              <span>
                数据源
              </span>
            </th>
            <td
              class="ant-descriptions-item-content"
              colspan="1"
            >
              <span>
                dataName(1)
              </span>
            </td>
          </tr>
          <tr
            class="ant-descriptions-row"
          >
            <th
              class="ant-descriptions-item-label"
              colspan="1"
            >
              <span>
                引擎版本
              </span>
            </th>
            <td
              class="ant-descriptions-item-content"
              colspan="1"
            >
              <span>
                1.12
              </span>
            </td>
          </tr>
          <tr
            class="ant-descriptions-row"
          >
            <th
              class="ant-descriptions-item-label"
              colspan="1"
            >
              <span>
                创建时间
              </span>
            </th>
            <td
              class="ant-descriptions-item-content"
              colspan="1"
            >
              <span>
                2022-01-01 00:00:00
              </span>
            </td>
          </tr>
          <tr
            class="ant-descriptions-row"
          >
            <th
              class="ant-descriptions-item-label"
              colspan="1"
            >
              <span>
                修改时间
              </span>
            </th>
            <td
              class="ant-descriptions-item-content"
              colspan="1"
            >
              <span>
                2022-01-01 00:00:00
              </span>
            </td>
          </tr>
          <tr
            class="ant-descriptions-row"
          >
            <th
              class="ant-descriptions-item-label"
              colspan="1"
            >
              <span>
                描述
              </span>
            </th>
            <td
              class="ant-descriptions-item-content"
              colspan="1"
            >
              <span>
                test
              </span>
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Test DetailInfo Component Test DetailInfo Component Should match snapshots 2`] = `
<DocumentFragment>
  <div
    class="ant-descriptions ant-descriptions-small ant-descriptions-bordered dt-taskinfo"
  >
    <div
      class="ant-descriptions-view"
    >
      <table>
        <tbody>
          <tr
            class="ant-descriptions-row"
          >
            <th
              class="ant-descriptions-item-label"
              colspan="1"
            >
              <span>
                函数名称
              </span>
            </th>
            <td
              class="ant-descriptions-item-content"
              colspan="1"
            >
              <span>
                <code>
                  test
                </code>
              </span>
            </td>
          </tr>
          <tr
            class="ant-descriptions-row"
          >
            <th
              class="ant-descriptions-item-label"
              colspan="1"
            >
              <span>
                类名
              </span>
            </th>
            <td
              class="ant-descriptions-item-content"
              colspan="1"
            >
              <span>
                className
              </span>
            </td>
          </tr>
          <tr
            class="ant-descriptions-row"
          >
            <th
              class="ant-descriptions-item-label"
              colspan="1"
            >
              <span>
                SQL
              </span>
            </th>
            <td
              class="ant-descriptions-item-content"
              colspan="1"
            >
              <span>
                sqlText
              </span>
            </td>
          </tr>
          <tr
            class="ant-descriptions-row"
          >
            <th
              class="ant-descriptions-item-label"
              colspan="1"
            >
              <span>
                用途
              </span>
            </th>
            <td
              class="ant-descriptions-item-content"
              colspan="1"
            >
              <span>
                purpose
              </span>
            </td>
          </tr>
          <tr
            class="ant-descriptions-row"
          >
            <th
              class="ant-descriptions-item-label"
              colspan="1"
            >
              <span>
                命令格式
              </span>
            </th>
            <td
              class="ant-descriptions-item-content"
              colspan="1"
            >
              <span>
                <code>
                  commandFormate
                </code>
              </span>
            </td>
          </tr>
          <tr
            class="ant-descriptions-row"
          >
            <th
              class="ant-descriptions-item-label"
              colspan="1"
            >
              <span>
                参数说明
              </span>
            </th>
            <td
              class="ant-descriptions-item-content"
              colspan="1"
            >
              <span>
                paramDesc
              </span>
            </td>
          </tr>
          <tr
            class="ant-descriptions-row"
          >
            <th
              class="ant-descriptions-item-label"
              colspan="1"
            >
              <span>
                创建
              </span>
            </th>
            <td
              class="ant-descriptions-item-content"
              colspan="1"
            >
              <span>
                2022-01-01 00:00:00
              </span>
            </td>
          </tr>
          <tr
            class="ant-descriptions-row"
          >
            <th
              class="ant-descriptions-item-label"
              colspan="1"
            >
              <span>
                最后修改
              </span>
            </th>
            <td
              class="ant-descriptions-item-content"
              colspan="1"
            >
              <span>
                2022-01-01 00:00:00
              </span>
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Test DetailInfo Component Test DetailInfo Component Should match snapshots 3`] = `
<DocumentFragment>
  <div
    class="ant-descriptions ant-descriptions-small ant-descriptions-bordered dt-taskinfo"
  >
    <div
      class="ant-descriptions-view"
    >
      <table>
        <tbody>
          <tr
            class="ant-descriptions-row"
          >
            <th
              class="ant-descriptions-item-label"
              colspan="1"
            >
              <span>
                资源名称
              </span>
            </th>
            <td
              class="ant-descriptions-item-content"
              colspan="1"
            >
              <span>
                test
              </span>
            </td>
          </tr>
          <tr
            class="ant-descriptions-row"
          >
            <th
              class="ant-descriptions-item-label"
              colspan="1"
            >
              <span>
                资源描述
              </span>
            </th>
            <td
              class="ant-descriptions-item-content"
              colspan="1"
            >
              <span>
                resourceDesc
              </span>
            </td>
          </tr>
          <tr
            class="ant-descriptions-row"
          >
            <th
              class="ant-descriptions-item-label"
              colspan="1"
            >
              <span>
                存储路径
              </span>
            </th>
            <td
              class="ant-descriptions-item-content"
              colspan="1"
            >
              <span>
                <code
                  class="cursor-pointer"
                >
                  url
                </code>
              </span>
            </td>
          </tr>
          <tr
            class="ant-descriptions-row"
          >
            <th
              class="ant-descriptions-item-label"
              colspan="1"
            >
              <span>
                创建
              </span>
            </th>
            <td
              class="ant-descriptions-item-content"
              colspan="1"
            >
              <span>
                userName 于2022-01-01 00:00:00
              </span>
            </td>
          </tr>
          <tr
            class="ant-descriptions-row"
          >
            <th
              class="ant-descriptions-item-label"
              colspan="1"
            >
              <span>
                修改时间
              </span>
            </th>
            <td
              class="ant-descriptions-item-content"
              colspan="1"
            >
              <span>
                2022-01-01 00:00:00
              </span>
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Test DetailInfo Component Test DetailInfo Component Should match snapshots 4`] = `
<DocumentFragment>
  <div
    class="ant-descriptions ant-descriptions-small ant-descriptions-bordered dt-taskinfo"
  >
    <div
      class="ant-descriptions-view"
    >
      <table>
        <tbody>
          <tr
            class="ant-descriptions-row"
          >
            <th
              class="ant-descriptions-item-label"
              colspan="1"
            >
              <span>
                名称
              </span>
            </th>
            <td
              class="ant-descriptions-item-content"
              colspan="1"
            >
              <span>
                mysql_test
              </span>
            </td>
          </tr>
          <tr
            class="ant-descriptions-row"
          >
            <th
              class="ant-descriptions-item-label"
              colspan="1"
            >
              <span>
                类型
              </span>
            </th>
            <td
              class="ant-descriptions-item-content"
              colspan="1"
            >
              <span>
                MySQL
              </span>
            </td>
          </tr>
          <tr
            class="ant-descriptions-row"
          >
            <th
              class="ant-descriptions-item-label"
              colspan="1"
            >
              <span>
                描述
              </span>
            </th>
            <td
              class="ant-descriptions-item-content"
              colspan="1"
            >
              <span>
                --
              </span>
            </td>
          </tr>
          <tr
            class="ant-descriptions-row"
          >
            <th
              class="ant-descriptions-item-label"
              colspan="1"
            >
              <span>
                连接信息
              </span>
            </th>
            <td
              class="ant-descriptions-item-content"
              colspan="1"
            >
              <span>
                <div>
                  <p
                    style="display: flex; line-height: 1.5; margin-bottom: 0px;"
                  >
                    <span
                      style="color: rgb(153, 153, 153); flex-shrink: 0;"
                    >
                      jdbcUrl：
                    </span>
                    <span
                      class="link-json"
                      title="jdbc:mysql://"
                    >
                      jdbc:mysql://
                    </span>
                  </p>
                  <p
                    style="display: flex; line-height: 1.5; margin-bottom: 0px;"
                  >
                    <span
                      style="color: rgb(153, 153, 153); flex-shrink: 0;"
                    >
                      用户名：
                    </span>
                    <span
                      class="link-json"
                      title="dtstack"
                    >
                      dtstack
                    </span>
                  </p>
                </div>
              </span>
            </td>
          </tr>
          <tr
            class="ant-descriptions-row"
          >
            <th
              class="ant-descriptions-item-label"
              colspan="1"
            >
              <span>
                连接状态
              </span>
            </th>
            <td
              class="ant-descriptions-item-content"
              colspan="1"
            >
              <span>
                <span
                  class="ant-badge ant-badge-status ant-badge-not-a-wrapper"
                >
                  <span
                    class="ant-badge-status-dot ant-badge-status-success"
                  />
                  <span
                    class="ant-badge-status-text"
                  >
                    正常
                  </span>
                </span>
              </span>
            </td>
          </tr>
          <tr
            class="ant-descriptions-row"
          >
            <th
              class="ant-descriptions-item-label"
              colspan="1"
            >
              <span>
                修改时间
              </span>
            </th>
            <td
              class="ant-descriptions-item-content"
              colspan="1"
            >
              <span>
                2022-11-02 03:00:42
              </span>
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Test DetailInfo Component Test DetailInfo Component Should match snapshots 5`] = `
<DocumentFragment>
  <div
    class="ant-descriptions ant-descriptions-small ant-descriptions-bordered dt-taskinfo"
  >
    <div
      class="ant-descriptions-view"
    >
      <table>
        <tbody>
          <tr
            class="ant-descriptions-row"
          >
            <th
              class="ant-descriptions-item-label"
              colspan="1"
            >
              <span>
                任务名称
              </span>
            </th>
            <td
              class="ant-descriptions-item-content"
              colspan="1"
            >
              <span>
                test
              </span>
            </td>
          </tr>
          <tr
            class="ant-descriptions-row"
          >
            <th
              class="ant-descriptions-item-label"
              colspan="1"
            >
              <span>
                实例ID
              </span>
            </th>
            <td
              class="ant-descriptions-item-content"
              colspan="1"
            >
              <span>
                <code
                  class="cursor-pointer"
                >
                  jobId
                </code>
              </span>
            </td>
          </tr>
          <tr
            class="ant-descriptions-row"
          >
            <th
              class="ant-descriptions-item-label"
              colspan="1"
            >
              <span>
                任务类型
              </span>
            </th>
            <td
              class="ant-descriptions-item-content"
              colspan="1"
            >
              <span>
                spark
              </span>
            </td>
          </tr>
          <tr
            class="ant-descriptions-row"
          >
            <th
              class="ant-descriptions-item-label"
              colspan="1"
            >
              <span>
                状态
              </span>
            </th>
            <td
              class="ant-descriptions-item-content"
              colspan="1"
            >
              <span>
                <span>
                  <div
                    class="dtc-circle-default dtc-circle-fail"
                  />
                    数据同步
                </span>
              </span>
            </td>
          </tr>
          <tr
            class="ant-descriptions-row"
          >
            <th
              class="ant-descriptions-item-label"
              colspan="1"
            >
              <span>
                调度周期
              </span>
            </th>
            <td
              class="ant-descriptions-item-content"
              colspan="1"
            >
              <span>
                <span>
                  分钟任务
                </span>
              </span>
            </td>
          </tr>
          <tr
            class="ant-descriptions-row"
          >
            <th
              class="ant-descriptions-item-label"
              colspan="1"
            >
              <span>
                计划时间
              </span>
            </th>
            <td
              class="ant-descriptions-item-content"
              colspan="1"
            >
              <span>
                1667358042000
              </span>
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
</DocumentFragment>
`;
